sys/syz-extract/netbsd.go

```diff
@@ -55,9 +56,20 @@ func (*netbsd) processFile(arch *Arch, info *compiler.ConstInfo) (map[string]uin
 	// try to extract consts with these prefixes as well.
 	compatNames := make(map[string][]string)
 	for _, val := range info.Consts {
-		compat := "LINUX_" + val
-		compatNames[val] = append(compatNames[val], compat)
-		info.Consts = append(info.Consts, compat)
+		const SYS = "SYS_"
+		if strings.HasPrefix(val, SYS) {
+			for _, prefix := range []string{"_", "__", "___"} {
+				for _, suffix := range []string{"30", "50"} {
```

Comment:
This hardcoding is invalid. syscalls with compat equivalents should be handled on individual basis.

Reply:
I’ll handle it as and when required. Right now it does the job. I’ll add individual cases later when we start finding conflicts.

Reply:
This is broken from the beginning. Even if it's temporarily papered out, it will break soon.

Reply:
If it works for the current set of syscalls, it's a pure accident.

Reply:
If you think there’s a better solution, please fork the repository and submit a pull request. Secondly, this code was always meant to work with the current set of syscalls. This code is always supposed to be written according to the syscalls in consideration. I’ll not make any changes until any conflicting syscall is added.

Reply:
As noted in the first, entry syscalls must be handled on individual basis. I object to merge this commit as is.

Reply:
@krytarowski What do you propose instead? An explicit map from "SYS_foo" to "SYS_foo50"?
How/when the current proposed solution will break?

Reply:
I propose to check if a syscall name is `SYS_semctl` and if so, rename it manually to `SYS_____semctl50`.

Long term it's good to add support to perform fuzzing on a range of supported NetBSD syscalls (from NetBSD-0.8 to HEAD). semctl has been updated 2 times.

```
$ grep semctl /usr/include/sys/syscall.h
/* syscall: "compat_14___semctl" ret: "int" args: "int" "int" "int" "union __semun *" */
#define SYS_compat_14___semctl  220
/* syscall: "compat_50_____semctl13" ret: "int" args: "int" "int" "int" "..." */
#define SYS_compat_50_____semctl13      301
/* syscall: "____semctl50" ret: "int" args: "int" "int" "int" "..." */
#define SYS_____semctl50        442
```

I think today it's enough to perform fuzzing just the current syscalls and skip compat ones.

```
$ grep _fstat /usr/include/sys/syscall.h|grep -v fstatfs|grep -v fstatvfs|grep -v statat
/* syscall: "compat_43_fstat43" ret: "int" args: "int" "struct stat43 *" */
#define SYS_compat_43_fstat43   62
/* syscall: "compat_12_fstat12" ret: "int" args: "int" "struct stat12 *" */
#define SYS_compat_12_fstat12   189
/* syscall: "compat_30___fstat13" ret: "int" args: "int" "struct stat13 *" */
#define SYS_compat_30___fstat13 279
/* syscall: "compat_50___fstat30" ret: "int" args: "int" "struct stat30 *" */
#define SYS_compat_50___fstat30 388
/* syscall: "__fstat50" ret: "int" args: "int" "struct stat *" */
#define SYS___fstat50   440
```

For example the `fstat` syscall can have 50 and 30 suffix + 43, 12, 13.

Reply:
First of all, this is not the place to rename syscalls. This only extracts syscalls and appends them to the list. If you want to have multiple semctl’s supported for different NetBSD versions then it requires proper designing on how to accomplish it.

Reply:
Naming syscalls as `__fstat50`, `compat_50___fstat30`, `compat_30___fstat13` right in the syscall descriptions looks reasonable to me. Especially since these are actually 3 different syscalls, with different arguments, each of which can be tested separately.

One downside is that if we have `fstat` syscall, we could figure out a scheme where take a corpus collected on linux and then use it on netbsd. This would allow us to progress slightly further on netbsd without coverage support. And we could also share descriptions of some syscall  between different OSes. However, if the syscalls have non-matching signatures (does signature of at least one netbsd fstat match that of linux?), then this may be not worth pursuing.

Reply:
I completely agree with Dmitry.

Reply:
Then let's go with this option and see how it will work for us over time. I think the root cause here is that we are doing somewhat strange things, so there is no clean/perfect solution.

Reply:
But until that happens, please merge this pull request so that we can perform fuzzing with the newer syscalls.

Reply:
None of our 'struct stat' is compatible with Linux, however this is rather well-defined POSIX interface, it should be able to test similar scenarios.

For now I propose to just keep using the newest syscall version in full-name in case there are compat options.

I think compat syscalls (old releases, compat_linux, netbsd32 (32-bit on 64-bit kernel)) are less important than getting kernel-ubsan / kernel-asan.

Reply:
So we agreed on __fstat50 as the way forward, good. I am going to merge this now as is as Utkarsh asks. I don't want to block progress by overthinking this particular aspect right now. Let's get back to this when we add more versions of fstat.